src/Utils.ml

```diff
@@ -58,3 +79,11 @@ let make_user_features feature_strings vars : (string * string) list =
                            let fdef = "(define-fun " ^ fname ^ sign ^ fs ^ ")"
                             in (fdef, fname))
    end
+
+let dedup_and_sort_keep_first ~compare list =
+  match list with
+  | [] -> []
+  | _ ->
+    let equal x x' = compare x x' = 0 in
+    let sorted = List.sort ~compare list in
+    List.remove_consecutive_duplicates ~which_to_keep:`First ~equal sorted
```

Comment:
Note that once you move this function into the List module above, you won't need the `List.` in front of `sort` (which should be changed to `stable_sort`) and `remove_consecutive_duplicates`.